### PR TITLE
BYC-1470 make nav component accessible

### PR DESCRIPTION
**Pull request by philt-symphony**
## Description

Nav was based on a nav, but used as a tablist. I changed it to be now a tablist and to be accessible following tab pattern:
https://www.w3.org/WAI/ARIA/apg/patterns/tabs/

## JIRA ticket

https://perzoinc.atlassian.net/browse/BYC-1470

## Demo

https://github.com/user-attachments/assets/17f6a43b-aa10-4e16-a594-106a267cfb31

### CODE REVIEW

**Review by benoit-sy — APPROVED** [review]

**Review by antoinerollindev — APPROVED** [review]

File: packages/components/src/components/nav/Nav.tsx [hunk 1]
```diff
@@ -24,6 +26,8 @@ export const Nav: React.FC<NavProps> = ({
   onActiveTabChange,
 }: NavProps) => {
 
+  const tabs = React.useRef([]);
```
[the remark this answers is not in the document]
Comment (philt-symphony) [reply packages/components/src/components/nav/Nav.tsx:29]: in the map when I create the buttons

[the remark this answers is not in the document]
Comment (philt-symphony) [reply packages/components/src/components/nav/Nav.tsx:29]: there https://github.com/SymphonyPlatformSolutions/symphony-ui-toolkit/pull/650/files#diff-59e04cc0647270fee50bd5ba4a1e57e9c7a2e3ef47893f8ee16f8cdb4723fad7R67

File: packages/components/src/components/nav/Nav.tsx [hunk 2]
```diff
@@ -39,20 +43,39 @@ export const Nav: React.FC<NavProps> = ({
...
-        </ul>
-      </nav>
+    <div role="tablist" className="tk-nav--tabs">
+      {items.map((item, idx) => (
+        <button
+          role = "tab"
+          ref={ref => {tabs.current[idx] = ref}}
+          key={item.id}
+          id={'tk-tab-' + item.id}
+          aria-controls={'tk-panel-' + item.panelId}
```
[the remark this answers is not in the document]
Comment (philt-symphony) [reply packages/components/src/components/nav/Nav.tsx:70]: Ah yes my bad I did some changes and I forgot to revert this part :)

File: packages/components/src/components/nav/Nav.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (philt-symphony) [reply packages/components/src/components/nav/Nav.tsx:29]: I amy need to test what happen if the props changes, maybe the ref will not reset

**Pull request merged**